vignettes/opt-dead-code.Rmd

```diff
@@ -30,6 +30,10 @@ speed_up <- function(obj) {
 
 # Dead Code Elimination
 
+## Idea
+
+Dead code elimination (also known as DCE, dead code removal, dead code stripping, or dead code strip) is an optimization to remove code which does not affect the program results. Removing such code has several benefits: it shrinks program size, an important consideration in some contexts, and it allows the running program to avoid executing irrelevant operations, which reduces its running time. It can also enable further optimizations by simplifying program structure. Dead code includes code that can never be executed (unreachable code), and code that only affects dead variables (written to, but never read again), that is, irrelevant to the program. 
```

Comment:
Oh, the idea behind the `Idea` section was to have an absolutely non-technical description of the optimizer.
this version is really similar to the `background` section.
that is why the student's PRs were so simplified. Please write something like "this optimizer will find chunks of code that would never be executed, or which execution won't influence the results...."

Reply:
Ohh!! I seemed to have missed the point ... 😅 ..
I'll fix it immediately...
Thanks :)

Reply:
Hello @jcrodriguez1989 , please have a look again. Now, I've used only the definitions used by the students and sewn them together to make it plausible as well as non-technical.
Thanks :D

Reply:
Hello @jcrodriguez1989 , please have a look again. Now, I've used only the definitions used by the students and sewn them together to make it plausible as well as non-technical.
Thanks :)